### PR TITLE
[libftp] Add new port

### PR DESCRIPTION
**Pull request by deniskovalchuk**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by deniskovalchuk** [comment]
> @deniskovalchuk please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

File: ports/libftp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(PACKAGE_NAME ftp
+                         CONFIG_PATH "share/cmake/ftp")
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
+configure_file("${CMAKE_CURRENT_LIST_DIR}/usage" "${CURRENT_PACKAGES_DIR}/share/${PORT}/usage" COPYONLY)
```
Comment (Mengna-Li) [inline ports/libftp/portfile.cmake:27]: `libftp` provides usage, please remove this and usage file.

Comment (Mengna-Li) [reply ports/libftp/portfile.cmake:27]: Add a new line at the end of the file.

**Review by Mengna-Li — CHANGES_REQUESTED** [review]

File: ports/libftp/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "libftp",
+  "version": "1.3.0",
+  "maintainers": "Denis Kovalchuk <denis.kovalchuk.main@gmail.com>",
+  "description": "A cross-platform FTP/FTPS client library based on Boost.Asio.",
+  "homepage": "https://github.com/deniskovalchuk/libftp",
+  "license": "MIT",
+  "supports": "!uwp",
```
Comment (Mengna-Li) [inline ports/libftp/vcpkg.json:8]: I noticed that you are the owner of this port. I did not see any description on the homepage of the port that does not support UWP. I would like to confirm if `libftp` doesn't support the UWP.

**Comment by Mengna-Li** [comment]
Usage tested pass on `x64-windows`.

File: ports/libftp/vcpkg.json (hunk 2 above)
Comment (deniskovalchuk) [reply ports/libftp/vcpkg.json:8]: Yes, this port isn't intended for UWP.

**Review by Mengna-Li — DISMISSED** [review]

File: ports/libftp/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,26 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (dg0yt) [inline ports/libftp/portfile.cmake:1]: DLL export woes?
```suggestion
if(VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()
```

Comment (deniskovalchuk) [reply ports/libftp/portfile.cmake]: Yes, I'm going to fix it in future releases.

**Review by Mengna-Li — DISMISSED** [review]

File: ports/libftp/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "libftp",
```
Comment (vicroms) [inline ports/libftp/vcpkg.json:2]: The name must [not match known packages when stripped of common prefixes and suffixes](https://learn.microsoft.com/vcpkg/contributing/maintainer-guide#check-names-against-other-repositories), in this case it collides with some packages and binaries found in Linux distributions like `ftplib` and `libftp4`. 

I suggest using `deniskovalchuk-libftp` as the port name.

**Review by vicroms — CHANGES_REQUESTED** [review]
Port name needs to be changed.

File: ports/libftp/vcpkg.json (hunk 4 above)
Comment (deniskovalchuk) [reply ports/libftp/vcpkg.json]: Fixed. Thanks!

**Review by vicroms — APPROVED** [review]

**Pull request merged**